### PR TITLE
[WIP] Update redhat updates test - updating from multiple versions

### PR DESCRIPTION
**Pull request by jkrocil**
Adds ability to run multiple version updates from a single yaml config.

There is also a workaround added for a form input element that is not present in versions below 5.2.2.3.

Collection example to help understand what I'm trying to achieve here:
```
<Module 'cfme/tests/configure/test_update_appliances.py'>
  <Function 'test_rhsm_direct[5.2.0.37 --> 5.2.5.2]'>
  <Function 'test_rhsm_rhn_mirror[5.2.0.37 --> 5.2.5.2]'>
  <Function 'test_sat5_direct[5.2.0.37 --> 5.2.5.2]'>
  <Function 'test_sat6_direct[5.2.0.37 --> 5.2.5.2]'>
  <Function 'test_sat6_rhn_mirror[5.2.0.37 --> 5.2.5.2]'>
  <Function 'test_rhsm_direct[5.2.2.3 --> 5.2.5.2]'>
  <Function 'test_rhsm_rhn_mirror[5.2.2.3 --> 5.2.5.2]'>
  <Function 'test_sat5_direct[5.2.2.3 --> 5.2.5.2]'>
  <Function 'test_sat6_direct[5.2.2.3 --> 5.2.5.2]'>
  <Function 'test_sat6_rhn_mirror[5.2.2.3 --> 5.2.5.2]'>
  <Function 'test_rhsm_direct[5.2.3.2 --> 5.2.5.2]'>
  <Function 'test_rhsm_rhn_mirror[5.2.3.2 --> 5.2.5.2]'>
  <Function 'test_sat5_direct[5.2.3.2 --> 5.2.5.2]'>
  <Function 'test_sat6_direct[5.2.3.2 --> 5.2.5.2]'>
  <Function 'test_sat6_rhn_mirror[5.2.3.2 --> 5.2.5.2]'>
  <Function 'test_rhsm_direct[5.2.4.2 --> 5.2.5.2]'>
  <Function 'test_rhsm_rhn_mirror[5.2.4.2 --> 5.2.5.2]'>
  <Function 'test_sat5_direct[5.2.4.2 --> 5.2.5.2]'>
  <Function 'test_sat6_direct[5.2.4.2 --> 5.2.5.2]'>
  <Function 'test_sat6_rhn_mirror[5.2.4.2 --> 5.2.5.2]'>
```

Expected runtime of this example would be about 5hrs btw (sat6 stuff is skipped).

__NOT TESTED YET (just the collection)__

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 8ea80e114501ee9bdf65dbf834f9dea249a5c2f0:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a7080a295b9cc08a356fe1b6dfa1a88586bbc7fc:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit e1b1ea235006357ffacfa172a0a369cb15bd649b:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6402fa4caade83f1ba33094eef039e448768e438:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/update_common.py:118:9: E265 block comment should start with '# '
./utils/db.py:15:1: F401 'lazycache' imported but unused
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4f3d43f24aef92bb484f144b77622301d3a538d3:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./utils/__init__.py:50:9: E265 block comment should start with '# '
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5d7f59dfbd2bd0a49355501bcca6e85d24ee5fd0:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./utils/__init__.py:50:9: E265 block comment should start with '# '
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5aceda5766cf27eecb3700c9710925b833d70a37:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 4af8a603de4909772d601abe2605bde471e70af4:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c8f6a2e62704e9c0ca5b1172ab3536c3b16da6a7:
:hurtrealbad: There were some flake issues that need to be resolved in order to merge the pull request:
```
./cfme/tests/update/test_update_sat6_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_cli_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_direct.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_rhsm_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat5_direct.py:16:1: F811 redefinition of unused 'appliance_set' from line 5
./cfme/tests/update/test_update_sat6_mirror.py:15:1: F811 redefinition of unused 'appliance_set' from line 5
```
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7fe79026b0bc6ddc60d671cd9040151ff1fa52c1:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d9b11b6d74b6b8bb7113707ececa97ef470aa30a:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 21a814a132492f685592b66a294c3a642e423328:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 71569cd7172774e8abe1fdfe0e66381746c7d5d5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 578b521eed4184ef47212a8cee29372eb8490e0d:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

File: cfme/configure/red_hat_updates.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def are_subscribed(*appliance_names):
         appliance_names: Names of appliances to check; will check all if empty
     """
     for row in get_appliance_rows(*appliance_names):
-        if row.update_status.text in {'Not Registered', 'Unsubscribed'}:
+        if row.update_status.text in {'Not registered', 'Unsubscribed'}:
```
Comment (mfalesni) [inline cfme/configure/red_hat_updates.py]: I'd personally `.lower()` the `.text` for matching (also on line 215)

**Comment by dajoRH** [comment]
Lint report for commit 6066321a429b11e0540f1bf6a4de9191b3124e46:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 751527638d84881fadf92f44c01720b938e59310:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
Lint report for commit 8d3bc758f174bf47eb44f5575bebb5e1b577ffad:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit a7cf071e2fdea70d602ba8e468e3405535a3b693:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 79ec245fdb9e2363942711cb993465dac308dd7c:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5c814c37cc96bfa25fd0c0d10a21187463a4ccee:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d88e621091456f205bce836365c8d65b9fcf821b:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 64d4e34c92d88cd1548170a33fd32378925b87ec:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
@jkrocil Tomorrow you have 6 months anniversary on this PR! :stuck_out_tongue:

**Comment by jkrocil** [comment]
Thanks

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 0e3fa0dcb9df1652f29d8547d1afc1be47d1cfe5:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6586d9497f28adb78c812b43b02b223f5d8473b8:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 6f7120969df610d905dfe52e93bfe3a1d0d12b33:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 5a9400a17ea9dc58f105780e898fd149de65c185:
:godmode: All seems good! :cake: :punch: :cookie: 
*CFME QE Bot*

**Comment by mfalesni** [comment]
7 months and still counting, man :stuck_out_tongue: :smile:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please?
*CFME QE Bot*

**Pull request closed without merging**